### PR TITLE
Improve the error message on pipeline create

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

If the pipeline is not found on pipeline create, it may be due to
a misspelled pipeline name or namespace name. Include both in the
error message to help troubleshooting.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [n/a] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by chmouel** [comment]
I think #311 is related would be nice if we get this fixed everywhere,

cc @danielhelfand

**Comment by afrittoli** [comment]
> I think #311 is related would be nice if we get this fixed everywhere,

I agree that it would be nice to fix other error messages, I was a bit lazy - only fixed the one I encountered :P 

About #311, I think the list and create commands are a bit different use cases. For the create, `tkn` looks for a specific pipeline specified via CLI parameters - and it may not find it - thus the *error* message.

For the list case, we don't expect anything specific, so if we don't get anything we can get a message that nothing was found, but that's for information only. To be honest my preference there would be to get no message at all, or if we do get a message get it on stderr - so that it's possible to reliably parse the stdout if needed.

File: pkg/cmd/pipeline/start.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ import (
 
 var (
 	errNoPipeline      = errors.New("missing pipeline name")
-	errInvalidPipeline = errors.New("invalid pipeline name")
+	errInvalidPipeline = "invalid pipeline name %s in namespace %s"
```
Comment (danielhelfand) [inline pkg/cmd/pipeline/start.go]: Would this read better as `pipeline name %s does not exist in namespace %s`?

**Comment by danielhelfand** [comment]
I think #311 should have a complete different error message that is specific to the namespace. This is more specific to the pipeline name not existing in a valid namespace, but I do think it more clearly communicates the issue to the user.

**Comment by chmouel** [comment]
let me know if that's good to go for you guys ?

File: pkg/cmd/pipeline/start.go (hunk 1 above)
Comment (afrittoli) [reply pkg/cmd/pipeline/start.go]: Fixed

**Comment by afrittoli** [comment]
I've been looking at different commands, and the errors returned a bit mixed up in style:

On pipeline commands:
* Start: only custom error (the one fixed in this PR)
* Delete: custom error + original client error
* Describe: original client error only
* Log: original client error only
* List: log a custom error and return the original error

On task commands:
* Delete: original client error
* List: log a custom error and return the original error

There are more commands for other resources too, I did not check them all; but it looks to me as if it would be best to agree on a common approach for the error message returned by commands, and then align them all. We could have one shared function used for logging for instance.

My proposal would be:
* normally only print a custom error message, which includes name and namespace of the resource
* in verbose mode (--verbose|-v) option passed, also print the original tekton client error

File: pkg/cmd/pipeline/start.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ import (
 
 var (
 	errNoPipeline      = errors.New("missing pipeline name")
-	errInvalidPipeline = errors.New("invalid pipeline name")
+	errInvalidPipeline = "pipeline name %s does not exist in namespace %"
```
Comment (piyush-garg) [inline pkg/cmd/pipeline/start.go]: nit: namespace %s

File: pkg/cmd/pipeline/start_test.go [hunk 3]
```diff
@@ -117,7 +117,7 @@ func Test_start_pipeline_not_found(t *testing.T) {
 
 	pipeline := Command(p)
 	got, _ := test.ExecuteCommand(pipeline, "start", "test-pipeline-2", "-n", "ns")
-	expected := "Error: " + errInvalidPipeline.Error() + "\n"
+	expected := "Error: " + fmt.Sprintf(errInvalidPipeline, "test-pipeline-2", "ns") + "\n"
```
Comment (piyush-garg) [inline pkg/cmd/pipeline/start_test.go]: its better to use exact message, not fmt.Sprintf, you may have test failed for the typo here if dont this way

File: pkg/cmd/pipeline/start.go [hunk 4]
```diff
@@ -367,7 +367,7 @@ func (opt *startOptions) startPipeline(pName string) error {
 	}
 
 	fmt.Fprintf(opt.stream.Out, "Pipelinerun started: %s\n\n"+
-		"In order to track the pipelinerun progress run:\ntkn pipelinerun logs %s -f\n", prCreated.Name, prCreated.Name)
+		"In order to track the pipelinerun progress run:\ntkn pipelinerun logs -n %s %s -f\n", prCreated.Name, prCreated.Namespace, prCreated.Name)
```
Comment (piyush-garg) [inline pkg/cmd/pipeline/start.go]: not sure, but I think `tkn pipelinerun logs name -f -n namespace` will give better readability

File: pkg/cmd/pipeline/start.go (hunk 2 above)
Comment (afrittoli) [reply pkg/cmd/pipeline/start.go]: upz, nice catch, thank you!

File: pkg/cmd/pipeline/start_test.go (hunk 3 above)
Comment (afrittoli) [reply pkg/cmd/pipeline/start_test.go]: heh, indeed

File: pkg/cmd/pipeline/start.go (hunk 4 above)
Comment (afrittoli) [reply pkg/cmd/pipeline/start.go]: done

**Pull request merged**